ch02/SmartSearch/SmartDictionary.cs

```diff
@@ -72,7 +72,7 @@ protected Rater RateItem(string search, Rater x)
         public IEnumerable<T> Search(string search, int maxItems)
         {
             if (search == null) throw new ArgumentNullException(nameof(search));
-            if (maxItems <=0) throw new ArgumentException(nameof(maxItems));
+            if (maxItems <=0) throw new ArgumentException($"{nameof(maxItems)} must be greater than zero", nameof(maxItems));
```

Comment:
I changed this because the name of the failing parameter is second parameter here.